lib/pf/Authentication/Source/LDAPSource.pm

```diff
@@ -196,10 +225,18 @@ sub _connect {
 
 sub match {
     my ($self, $params) = @_;
+    my $start = Time::HiRes::gettimeofday();
     if($self->is_match_cacheable) {
-        return $self->cache->compute([$self->id, $params], sub { return $self->SUPER::match($params)});
+        my $result = $self->cache->compute([$self->id, $params], sub {
+                $pf::StatsD::statsd->increment(called() . "." . $self->id. ".cache_miss.count" );
+                my $result =   $self->SUPER::match($params);
```

Comment:
Good catch.
This code is getting a bit complex.
Perhaps time to extract the nameless sub and give it a proper name.